src/cascadia/TerminalApp/CascadiaSettings.h

```diff
@@ -99,7 +99,7 @@ class TerminalApp::CascadiaSettings final
     Json::Value _userDefaultProfileSettings{ Json::Value::null };
 
     void _LayerOrCreateProfile(const Json::Value& profileJson);
-    Profile* _FindMatchingProfile(const Json::Value& profileJson);
+    winrt::com_ptr<winrt::TerminalApp::implementation::Profile> _FindMatchingProfile(const Json::Value& profileJson);

```

Comment:
is it necessary for this one to return an implementation type? We were only returning a raw pointer before because we wanted `null`ness

Reply:
We use this in `CascadiaSettings::_LayerOrCreateProfile(Json)`. Since the Json-aware functions are all on the impl type, the first thing I'd have to do is convert it from the projected type to the impl type. So figured I'd save us an extra step (especially since we already have the impl type on the way out of `_FindMatchingProfile`).

In general, my philosophy is going to be that all Json stuff will deal with the impl type. We could investigate breaking this up further but that's probably something for later.